src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size + 1) {
+            throw new ArrayListIndexOutOfBoundsException("Index "
+                    + index + "is out of bounds " + size);
+        }
+        if (size == elementData.length || index > size) {
+            elementData = grow();
+        }
+        T[] elemAfterIndex = (T[]) new Object[size - index + 1];
+        System.arraycopy(elementData,index,elemAfterIndex,0,size - index + 1);
+        elementData[index] = value;
+        if (index <= size) {
+            System.arraycopy(elemAfterIndex, 0, elementData, index + 1, elemAfterIndex.length - 1);
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            T elem = (T) list.get(i);
+            add(elem);
+        }
+    }
 
+    private T[] grow() {
+        int oldCapacity = elementData.length;
+        int newCapacity = oldCapacity + (oldCapacity >> 1);
+        T[] newElementData = (T[]) new Object[newCapacity];
+        System.arraycopy(elementData,0, newElementData,0, oldCapacity);
```

Comment:
autoformat this file, should be a space after each comma (Ctrl Alt L to call IDEA's autoformat on Windows machines)